### PR TITLE
Add circleci & codecov integration

### PR DESCRIPTION
**Pull request by bhazelton**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - restore_cache:
+          key: deps-{{ .Branch }}-{{ checksum ".circleci/puget_tests.yml" }}
+      - run:
+          name: configure conda environment
+          command: ./.circleci/install-circle.sh
+      - run:
+          name: install
+          command: |
+            source activate ${ENV_NAME}
+            pip install recordlinkage
```
Comment (arokem) [inline .circleci/config.yml]: Doesn't this one get picked up by the `pip install .`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            pip install .
+      - run:
+          name: run puget tests
+          command: |
+            source activate ${ENV_NAME}
+            mkdir test-reports
+            python -m pytest puget -v --cov=puget --cov-config=.coveragerc --cov-report xml:./coverage.xml --junitxml=test-reports/xunit.xml
```
Comment (arokem) [inline .circleci/config.yml]: I see that codecov uses the coverage.xml output. Who uses the xunit.xml output?

File: .circleci/config.yml (hunk 1 above)
Comment (bhazelton) [reply .circleci/config.yml]: It probably does, I was being explicit. I can take it out.

File: .circleci/config.yml (hunk 2 above)
Comment (bhazelton) [reply .circleci/config.yml]: I'm not sure, maybe no one. This was something I copied from some other project for another repo and just copied here too. I can take it out and find out what happens.

File: .circleci/config.yml (hunk 1 above)
Comment (arokem) [reply .circleci/config.yml]: Probably best, so you don't end up wondering about this down the line.

File: .circleci/config.yml (hunk 2 above)
Comment (arokem) [reply .circleci/config.yml]: The ol' cargo cult of OSS engineering.

Comment (bhazelton) [reply .circleci/config.yml]: I took it out and everything worked, so seems ok.

**Comment by bhazelton** [comment]
I use circleci on another project and am *very* happy with it.

**Pull request merged**